Support sound effect

ChinYikMing opened the pull request:
Use the newly introduced sound related system calls setup_audio and control_audio to play sound effects. No music support yet because Quake requires CD-ROM.

jserv commented:
> No music support yet because Quake requires CD-ROM.

Did you mean that there is no music support in shareware data package?

jserv commented on port/snd.c:
```diff
@@ -0,0 +1,206 @@
+/*
+Copyright (C) 1996-1997 Id Software, Inc.
+
```
Add your formal name as well.

jserv commented on winquake/sound.h:
```diff
@@ -36,17 +36,19 @@ typedef struct sfx_s
 	int 	width;
 	int 	stereo;
-	byte	data[1];		// variable sized
+	//byte	data[1];		// variable sized
```
Don't leave the dead code.

ChinYikMing commented:
> > No music support yet because Quake requires CD-ROM.
> 
> Did you mean that there is no music support in shareware data package?

I think so because I do not see any music or bgm related files in the extracted pak file.

jserv commented:
> > Did you mean that there is no music support in shareware data package?
> I think so because I do not see any music or bgm related files in the extracted pak file.

Later, you should update `docs/demo.md` (rv32emu) to address these concerns.

jserv commented on port/snd.c:
```diff
@@ -0,0 +1,206 @@
+sfxcache_t *S_LoadSound (sfx_t *sfx)
+{
+    char buf[64];
+    strcpy(buf, "sound/");
```
Use `sprintf` instead.

jserv commented on port/snd.c:
```diff
@@ -0,0 +1,206 @@
+    sfx = malloc(sizeof(sfx_t));
+    if(!sfx){
+        Con_Printf("S_CacheSound malloc failed\n");
+    return NULL;
```
Indent here.

jserv commented on winquake/sound.h:
```diff
@@ -36,17 +36,18 @@ typedef struct sfx_s
 // !!! if this is changed, it much be changed in asm_i386.h too !!!
 typedef struct
 {
+	void    *data;
```
Is there any strong reason to change the order of `data` member?

ChinYikMing replied:
> Is there any strong reason to change the order of `data` member?

Data and size in applications must be placed at first two fields in structure so that it makes emulator compatible to any applications(32-bit / 64-bit) when accessing different sfxinfo_t from applications

jserv replied:
Mention this in git commit messages along with proper explanation.

ChinYikMing committed:
Support sound effect

Use the newly introduced sound related system calls
setup_audio and control_audio to play sound effects.
No music support yet because Quake requires CD-ROM.

To make the rv32emu compatible with all applications (32-bit / 64-bit),
sound data and size must be supplied in the first two elements
of the structure (sfxcache_t in Quake).

jserv commented:
Thank @ChinYikMing for contributing!
